Populate our pre-image to the ValidatorSet after catchup

Geod24 opened the pull request:
```
One scenario that was missing in PR2205 was the state right after construction.
Because onAcceptedBlock is not called when replaying blocks (or starting from scratch),
to avoid blocks being re-written to disk and stats being wrong,
the initial state was still faulty. It is now fixed.
```

CC @hewison-chris

Geod24 committed:
Populate our pre-image to the ValidatorSet after catchup

One scenario that was missing in PR2205 was the state right after construction.
Because onAcceptedBlock is not called when replaying blocks (or starting from scratch),
to avoid blocks being re-written to disk and stats being wrong,
the initial state was still faulty. It is now fixed.

hewison-chris reviewed (approved):
LGTM

hewison-chris commented:
Looks like some tests fail though.

kchulj reviewed (approved):
LGTM
The CI fails, as @hewison-chris mentioned.

Geod24 committed:
Remove a flakey test that relies on PreImage not being propagated

This test has an assert which checks that a node doesn't know its own pre-image,
and then that the API receive less than 100 (not 400, as documented) calls.
A node should always know its own pre-image, and as we are implementing better
gossiping for pre-image, the test will spuriously (or reliably) fail.
The very basis of this test (testing that an endpoint is called less than X times)
is dubious in itself, as it doesn't test an external behavior of the system.

Geod24 commented:
Found the "bug": There is a test that test explicitly that pre-images aren't propagated. Removed it, with explanations.

Geod24 commented:
Can I have a re-review ?

linked0 reviewed (approved):
LGTM
